x-pack/solutions/observability/plugins/apm/public/components/shared/environment_select/index.tsx

```diff
@@ -53,60 +68,80 @@ export function EnvironmentSelect({
   onChange: (value: string) => void;
 }) {
   const [searchValue, setSearchValue] = useState('');
-
-  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
-
-  const selectedOptions: Array<EuiComboBoxOptionOption<string>> = [
+  const [selectedOption, setSelectedOption] = useState<Array<EuiComboBoxOptionOption<string>>>([
     {
       value: environment,
       label: getEnvironmentLabel(environment),
     },
-  ];
+  ]);
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+  const { data, onSearchChange, searchStatus } = useEnvironmentSelect({ serviceName, start, end });
 
   const onSelect = (changedOptions: Array<EuiComboBoxOptionOption<string>>) => {
     if (changedOptions.length === 1 && changedOptions[0].value) {
       onChange(changedOptions[0].value);
     }
+    setSelectedOption(changedOptions);
   };
 
-  const { data, status: searchStatus } = useFetcher(
-    (callApmApi) => {
-      return isEmpty(searchValue)
-        ? Promise.resolve({ terms: [] })
-        : callApmApi('GET /internal/apm/suggestions', {
-            params: {
-              query: {
-                fieldName: SERVICE_ENVIRONMENT,
-                fieldValue: searchValue,
-                serviceName,
-                start,
-                end,
-              },
-            },
-          });
+  const terms = useMemo(() => {
+    if (isEmpty(searchValue)) {
```

Comment:
Nit: same
```suggestion
    if (searchValue !== '')) {
```
or just `searchValue`